hal/armv8m/mcx/interrupts.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Interrupt handling
+ *
+ * Copyright 2017, 2020, 2022, 2024 Phoenix Systems
+ * Author: Pawel Pisarczyk, Hubert Buczynski, Damian Loewnau, Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "hal/interrupts.h"
+#include "hal/spinlock.h"
+#include "hal/cpu.h"
+#include "hal/list.h"
+#include "hal/armv8m/armv8m.h"
+#include "hal/armv8m/mcx/n94x/mcxn94x.h"
+
+#include "proc/userintr.h"
+
+#include "config.h"
+
+/* Value based on other target architectures */
+#define SIZE_HANDLERS 4
```

Comment:
Unused define, I guess it's a leftover that wasn't removed (present in most irq drivers). General comment: it'd be nice to share the NVIC code across CPUs using it, but that's not in scope of this PR.